pkg/dispatcher/prometheus_test.go

```diff
@@ -0,0 +1,205 @@
+package dispatcher
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+type prometheusAPIForTest struct {
+	queryFunc func(ctx context.Context, query string, ts time.Time) (model.Value, v1.Warnings, error)
+}
+
+func (prometheusAPI *prometheusAPIForTest) Query(ctx context.Context, query string, ts time.Time) (model.Value, v1.Warnings, error) {
+	return prometheusAPI.queryFunc(ctx, query, ts)
```

Comment:
Can we make this validate the `query string` parameter?

Reply:
Done.